agent/api/task/task_test.go

```diff
@@ -4272,11 +4287,11 @@ func TestTaskServiceConnectAttachment(t *testing.T) {
 						AttachmentArn: strptr("attachmentArn"),
 						AttachmentProperties: []*ecsacs.AttachmentProperty{
 							{
-								Name:  strptr(serviceConnectConfigKey),
+								Name:  strptr(serviceconnect.ServiceConnectConfigKey),
 								Value: strptr(tc.testSCConfigValue),
 							},
 							{
-								Name:  strptr(serviceConnectContainerNameKey),
+								Name:  strptr(serviceconnect.ServiceConnectContainerNameKey),
```

Comment:
This test usage feels like a week reason to adjust visibility of these values with the s vs S beginning

Reply:
Added getters. I thought about hardcoding them in this test file but I'm not sure if I like the idea that changing the internal service connect config keys will break this unit test when it shouldn't really matter.

Reply:
That works, or you can have a function accessible in the test realm. Specifically `service_connect_attachment_parser_test.go` can have a helper `CreateServiceConnectAttachment` or something with arguments that can be passed in and returning a correctly key'd map to merge here. That function only exists in test and doesn't require getters or exporting and can provide a reminder when changing things in the attachment parser that you have to support that use-case

Reply:
I couldn't seem to get it to work. It appears that test files are compiled separately for each package. According to https://pkg.go.dev/cmd/go/internal/test -`"Each listed package causes the execution of a separate test binary."` For that reason, test files could not access even exported functions from a test file in another package.

Reply:
TIL - thanks.